sageresearch-app-sdk/src/main/java/org/sagebionetworks/research/sageresearch/reminders/ReminderAlarmReceiver.kt

```diff
@@ -0,0 +1,207 @@
+import android.content.Intent
+import android.media.RingtoneManager
+import android.net.Uri
+import android.os.Build
+import android.support.v4.app.NotificationCompat
+import org.researchstack.backbone.ui.MainActivity
+import org.sagebionetworks.research.sageresearch_app_sdk.R
+
+import org.slf4j.LoggerFactory
+
+/**
+ * The ReminderAlarmReceiver can be sub-classed and used to easily control showing notifications to the user
+ */
+open class ReminderAlarmReceiver: BroadcastReceiver() {
+
+    private val logger = LoggerFactory.getLogger(ReminderAlarmReceiver::class.java)
+
+    /**
+     * @property notificationChannelId set on Android OS >= 26, uniquely identified your notifications
+     */
+    protected open val notificationChannelId: String
+        get() {
+            return "Sage-Research NotificationChannelId"
+        }
+
+    /**
+     * @property notificationChannelId set on Android OS >= 26, explains your notifications
+     */
+    protected open val notificationChannelTitle: String
+        get() {
+            return "Sage-Research Reminders"
+        }
+
+    /**
+     * @property notificationChannelId set on Android OS >= 26, explains your notifications in more detail
+     */
+    protected open val notificationChannelDesc: String
+        get() {
+            return "SageResearch reminders help you remember to log your data."
+        }
+
+    /**
+     * If true, when the notification is tapped, it will leave the user's status bar, false, it will stay there
+     */
+    protected open val shouldAutoCancel: Boolean
+        get() {
+            return true
+        }
+
+    /**
+     * @param code associated with this sound
+     * @param action of the notification
+     * @return an alarm sound that plays when the notification appears in the status bar
+     */
+    protected open fun alarmSound(code: Int, action: String): Uri? {
+        return RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION)
+    }
+
+    /**
+     * @param code associated with this icon
+     * @param action of the notification
+     * @return a drawable resource value of the small icon that shows with the notification
+     */
+    protected open fun notificationIcon(code: Int, action: String): Int? {
+        return R.mipmap.ic_launcher
+    }
+
+    /**
+     * @param will be the broadcast service context
+     * @param code associated with this Activity to run when the notification is tapped
+     * @param action of the notification
+     * @return the pending intent which is run when the user taps the notification
+     */
+    protected open fun pendingIntent(context: Context, code: Int, action: String): PendingIntent {
+        val notificationIntent = Intent(context, MainActivity::class.java)
+        notificationIntent.action = action
+        notificationIntent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP
+        val stackBuilder = TaskStackBuilder.create(context)
+        stackBuilder.addParentStack(MainActivity::class.java)
```

Comment:
Do we need to build flexibility around this for non ResearchStack tasks? Or does the task stack not really matter?

Reply:
The task stack does matter.  There is flexibility already built in because this function is open to sub-classes.